yt_dlp/cookies.py

```diff
@@ -0,0 +1,738 @@
+import ctypes
+import json
+import os
+import shutil
+import sqlite3
+import struct
+import subprocess
+import sys
+import warnings
+from datetime import datetime, timedelta
+
+from yt_dlp.aes import aes_cbc_decrypt
+from yt_dlp.compat import compat_cookiejar_Cookie, compat_b64decode, compat_TemporaryDirectory
+from yt_dlp.utils import YoutubeDLCookieJar, expand_path, bytes_to_intlist, intlist_to_bytes
+
+try:
+    from Crypto.Cipher import AES
+    CRYPTO_AVAILABLE = True
+except ImportError:
+    CRYPTO_AVAILABLE = False
+
+try:
+    import keyring
+    KEYRING_AVAILABLE = True
+except ImportError:
+    KEYRING_AVAILABLE = False
+
+
+SUPPORTED_BROWSERS = ['brave', 'chrome', 'chromium', 'edge' 'firefox', 'opera', 'safari', 'vivaldi']
+CHROMIUM_BASED_BROWSERS = {'brave', 'chrome', 'chromium', 'edge' 'opera', 'vivaldi'}
```

Comment:
the supported browsers list is ordered because it's used to display the list in the `--help` text. Storing separately and as sets is fine but I'll sort them before printing them in `parseOpts()`